### ISSUE TITLE
Need a way to disable "Content-Encoding:gzip" response header for CBL

### ISSUE THREAD

**Issue by snej** [couchbase/sync_gateway#1057]
**tl;dr:** CBL/iOS needs a way to get SG to _not_ add a `Content-Encoding: gzip` header when it serves an attachment that's stored in gzip-encoded form.

When CBL lazily pulls an attachment it does a GET of the attachment's URL from SG. Now, some attachments are stored in gzipped form, identified by `"encoding":"gzip"` in their metadata. When the GET handler serves such an attachment it sets the HTTP response header `Content-Encoding: gzip` to identify the response as being zipped.

The problem is that when the Cocoa codebase downloads such an attachment, NSURLConnection automatically recognizes the Content-Encoding header and unzips the data before we can access it. That means we can't store the attachment in its original form — the best we can do is gzip it again while storing it. Unfortunately that makes it likely that the SHA-1 digest won't match, if the gzip settings aren't identical to the ones used to compress it originally.

(It's possible other platforms' HTTP client libraries have this same issue, since most of the time it's very convenient to have Content-Encoding decoded for you.)

I'm working around this by defining a `?content_encoding=false` URL query parameter on the request that instructs SG to omit the Content-Encoding header.
